### PR TITLE
feat!: Support php 8.2 vol 4766

### PR DESCRIPTION
**Pull request by fibble**
## Description

PHP ^8.2 support

Related issue: [VOL-4766](https://dvsa.atlassian.net/browse/VOL-4766)

Breaking Change - requires new minimum PHP Version.

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [x] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "url": "https://github.com/dvsa/laminas-form.git"
     }],
     "require": {
-        "php": "~8.0.0",
+        "php": "^8.2",
```
Comment (JoshuaLicense) [inline composer.json:9]:
```suggestion
        "php": "~8.2.0",
```

File: composer.json [hunk 2]
```diff
@@ -50,7 +50,8 @@
         "mockery/mockery": "^1.6",
         "phpunit/phpunit": "^9.6",
         "san/san-session-toolbar": "*",
-        "bamarni/composer-bin-plugin": "^1.8"
+        "bamarni/composer-bin-plugin": "^1.8",
+        "rector/rector": "^1.0"
```
Comment (JoshuaLicense) [inline composer.json:54]: Should we keep this out of the main dependency tree for now?

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]
Looks good, few suggestions left.

File: composer.json (hunk 2 above)
Comment (fibble) [reply composer.json:54]: Yeah i was going to remove that but missed it thanks.

File: composer.lock [hunk 3]
```diff
@@ -7625,6 +7543,65 @@
             ],
             "time": "2024-04-05T04:35:58+00:00"
         },
+        {
+            "name": "rector/rector",
+            "version": "1.0.5",
+            "source": {
+                "type": "git",
```
Comment (JoshuaLicense) [inline composer.lock:7550]: Still contains `rector`. Can we update this lockfile too, please.

Comment (fibble) [reply composer.lock]: Of course - sorted. thanks.

**Review by JoshuaLicense — APPROVED** [review]

**Pull request merged**